dev/benchmarks/macrobenchmarks/lib/web_benchmarks.dart

```diff
@@ -347,6 +381,20 @@ class LocalBenchmarkServerClient {
     );
   }
 
+  /// Reports a message about the demo to the benchmark server.
+  Future<void> reportDemo(String report) async {
```

Comment:
The word "demo" is too specific to the gallery to be used in this file. Let's give this a more generic terminology, maybe `printToConsole` for method/function names and `consolePrinter` for objects/closures. Instead of "report" I'd use "message".

Reply:
`printToConsole (String message)` and `/print-to-console` look good to me.
The gallery recorder can be constructed as:
```dart
  GalleryRecorder(
    benchmarkName: '${_galleryBenchmarkPrefix}_animated_widgets_perf',
    consolePrinter: _client.printToConsole,
    errorPrinter: _client.reportError,
    runCriterion: (String demo) =>
    typeOfDemo(demo) == DemoType.animatedWidget,
  )
```

Reply:
We shouldn't need to pass `consolePrinter` and `errorPrinter` to `GalleryRecorder`. It should be just:

```
  GalleryRecorder(
    benchmarkName: '${_galleryBenchmarkPrefix}_animated_widgets_perf',
    runCriterion: (String demo) =>
    typeOfDemo(demo) == DemoType.animatedWidget,
  )
```

Reply:
Updated to remove `reporter` and `reportError` for recorders.